### PR TITLE
Add availabilities model

### PR DESCRIPTION
**Pull request by insoochung**
- Adds availabilities model
- Routes to be added to the API doc
```
                           availabilities GET    /availabilities(.:format)                                                                         availabilities#index
                                          POST   /availabilities(.:format)                                                                         availabilities#create
                             availability GET    /availabilities/:id(.:format)                                                                     availabilities#show
                                          PATCH  /availabilities/:id(.:format)                                                                     availabilities#update
                                          PUT    /availabilities/:id(.:format)                                                                     availabilities#update
                                          DELETE /availabilities/:id(.:format)                                                                     availabilities#destroy
```

### CODE REVIEW

File: server/app/controllers/availabilities_controller.rb [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+  end
+
+  # POST /availabilities/
+  def create
+    if !confirm_requester_is_admin_or_rep_or_volunteer
+      return
+    end
+    params = availability_params.to_h
+    # If owner key is not provided, use the owner key provided by the requester
+    if !params.key?("user_id")
```
Comment (ryanzesch) [inline server/app/controllers/availabilities_controller.rb:36]: Just to be sure - does this allow any admin/rep/volunteer to create availability for any other user? (I don't think that's the intended functionality)

Comment (ryanzesch) [reply server/app/controllers/availabilities_controller.rb:36]: + Same question for the other routes

**Review by ryanzesch — COMMENTED** [review]
Looks good apart from the one comment

File: server/app/controllers/availabilities_controller.rb (hunk 1 above)
Comment (insoochung) [reply server/app/controllers/availabilities_controller.rb:36]: Good catch should fix this I'll render an error if it's not an admin

Comment (insoochung) [reply server/app/controllers/availabilities_controller.rb:36]: For other routes, you should be able to get, update, delete availability only if you are admin or an owner of that availability

**Review by ryanzesch — APPROVED** [review]
Looks good!

**Pull request merged**